main_window/rocket_data.py

```diff
@@ -157,19 +157,22 @@ def last_value_and_time(self, device: DeviceType, data_entry_id: DataEntryIds) -
         :return: Value, Time
         :rtype:
         """
-        ret = self.time_series_by_device(device, data_entry_id)
-
-        if ret is None:
-            return None
+        with self.data_lock:
+            all_times = list(self.timeset.keys())
 
-        t, y = ret
+            full_address = self.device_manager.get_full_address(device)
+            if full_address is None:
+                return None
 
-        if len(t) > 0:
-            return y[-1], t[-1]
+            data_entry_key = DataEntryKey(full_address, data_entry_id)
+            # iterate in reverse time order to get most recent entry
+            for time in reversed(all_times):
```

Comment:
Python dictionaries support iterating over the values only, so this for loop can probably be simplified by choosing a different thing to iterate over